src/main/java/io/github/fablabsmc/fablabs/impl/fiber/annotation/BackedConfigLeaf.java

```diff
@@ -0,0 +1,161 @@
+
+	public BackedConfigLeaf(ConfigLeaf<S> backing, ConfigType<R, S, ?> type, Object pojo, Field backingField) {
+		this.backing = backing;
+		this.type = type;
+		this.pojo = pojo;
+		this.backingField = backingField;
+	}
+
+	@Override
+	public String getName() {
+		return backing.getName();
+	}
+
+	@Override
+	public Map<FiberId, ConfigAttribute<?>> getAttributes() {
+		return backing.getAttributes();
+	}
+
+	@Override
+	public <R, A> Optional<R> getAttributeValue(FiberId id, ConfigType<R, A, ?> type) {
+		return backing.getAttributeValue(id, type);
+	}
+
+	@Override
+	public <A> Optional<A> getAttributeValue(FiberId id, SerializableType<A> expectedType) {
+		return backing.getAttributeValue(id, expectedType);
+	}
+
+	@Override
+	public <A> ConfigAttribute<A> getOrCreateAttribute(FiberId id, SerializableType<A> attributeType, @Nullable A defaultValue) {
+		return backing.getOrCreateAttribute(id, attributeType, defaultValue);
+	}
+
+	@Nullable
+	@Override
+	public ConfigBranch getParent() {
+		return parent;
+	}
+
+	@Override
+	public void attachTo(ConfigBranch parent) throws IllegalTreeStateException {
+		// Attaching/detaching requires careful inspection of whether or not the parent's items contains `this`.
+		// In the case of deferred leaves, the leaf in the parent's items is the deferred leaf and not the backing leaf.
+		// Thus, we can't defer attaching/detaching to the backing leaf, as it will think it's not part of the parent's items.
+		if (this.parent != null && this.parent != parent) {
+			throw new IllegalTreeStateException(this + " needs to be detached before changing the parent");
+		}
+
+		// this node may have already been added by the collection
+		if (parent != null && !parent.getItems().contains(this)) {
+			parent.getItems().add(this);
+		}
+
+		this.parent = parent;
+	}
+
+	@Override
+	public void detach() {
+		// Note: infinite recursion between ConfigNode#detach() and NodeCollection#remove() could occur here,
+		// but the latter performs the actual collection removal before detaching
+		if (this.parent != null) {
+			// here, we also need to avoid triggering a ConcurrentModificationException
+			// thankfully, remove does not cause a CME if it's a no-op
+			this.parent.getItems().remove(this);
+		}
+
+		this.parent = null;
+	}
+
+	@Override
+	public boolean setValue(@Nonnull S value) {
+		return backing.setValue(value);
+	}
+
+	@Nonnull
+	@Override
+	@SuppressWarnings("unchecked")
+	public S getValue() {
+		try {
+			backingField.setAccessible(true);
+			R fieldValue = (R) backingField.get(pojo);
+
+			if (!Objects.equals(fieldValue, cachedValue)) {
+				this.setValue(type.toSerializedType(fieldValue));
+				cachedValue = fieldValue;
+			}
+		} catch (IllegalAccessException e) {
+			// Because this exception might appear to happen 'at random' to the user, we wrap it to at least provide more information about what just happened
+			(new RuntimeFiberException("Couldn't fetch setting value from POJO", e)).printStackTrace();
```

Comment:
This exception should be rethrown, if it ever happens it indicates a bug on our side.